pylint/checkers/typecheck.py

```diff
@@ -499,16 +499,7 @@ def _emit_no_member(
                 return False
         except astroid.NotFoundError:
             return True
-    if (
-        owner.parent
-        and isinstance(owner.parent, nodes.ClassDef)
-        and owner.parent.name == "EnumMeta"
-        and owner_name == "__members__"
-        and node.attrname in {"items", "values", "keys"}
-    ):
-        # Avoid false positive on Enum.__members__.{items(), values, keys}
-        # See https://github.com/PyCQA/pylint/issues/4123
-        return False
```

Comment:
I saw ln 511 was uncovered and I investigated further. Turns out new code additions after this was added make it that this case is no longer necessary as the code short-circuits early. This is proven by the fact that functional tests under member_checks such as `for itm in Animal.__members__.items():` still pass.